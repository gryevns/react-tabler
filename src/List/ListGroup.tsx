import * as React from "react";
import cn from "classnames";

export interface ListGroupProps {
    className?: string;
    children: React.ReactNode;
    transparent?: boolean;
    card?: boolean;
}

function ListGroup({ className, children, transparent, card }: ListGroupProps) {
    const classes = cn(
        "list-group",
        "mb-0",
        {
            "list-group-transparent": transparent,
            "card-list-group": card,
        },
        className,
    );
    return <div className={classes}>{children}</div>;
}

export default ListGroup;
