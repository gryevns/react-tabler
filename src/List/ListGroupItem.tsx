import * as React from "react";
import cn from "classnames";

export interface ListGroupItemProps {
    className?: string;
    children: React.ReactNode;
    as?: React.ElementType;
    active?: boolean;
    action?: boolean;
    icon?: string;
    to?: string;
}

function ListGroupItem({
    className,
    children,
    as = "a",
    active,
    action,
    icon,
}: ListGroupItemProps) {
    const classes = cn(
        "list-group-item",
        {
            "list-group-item-action": action,
        },
        {
            active: active,
        },
        className,
    );
    const Component = as;
    return (
        <Component className={classes}>
            {icon && <span className="mr-3 icon">TODO</span>}
            {children}
        </Component>
    );
}

export default ListGroupItem;
