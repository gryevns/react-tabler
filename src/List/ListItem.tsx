import * as React from "react";
import cn from "classnames";

interface ListItemProps {
    className?: string;
    children: React.ReactNode;
    inline?: boolean;
    separated?: boolean;
}

function ListItem({ className, children, inline, separated }: ListItemProps) {
    const classes = cn(
        { "list-inline-item": inline, "list-separated-item": separated },
        className,
    );
    return <li className={classes}>{children}</li>;
}

export default ListItem;
