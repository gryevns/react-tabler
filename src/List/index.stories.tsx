import React from "react";
import { Trash } from "react-feather";

import Avatar from "../Avatar";
import Card from "../Card";
import List from "../List";

export default {
    title: "List",
};

export const Default = () => (
    <List>
        <List.Item>An Item</List.Item>
        <List.Item>Another Item</List.Item>
        <List.Item>A third item</List.Item>
    </List>
);

export const Unstyled = () => (
    <List unstyled>
        <List.Item>An Item</List.Item>
        <List.Item>Another Item</List.Item>
        <List.Item>A third item</List.Item>
    </List>
);

export const Styled = () => (
    <Card>
        <Card.Header>
            <Card.Title>Members</Card.Title>
        </Card.Header>
        <Card.Body className="overflow-auto" style={{ height: "15rem" }}>
            <List unstyled seperated>
                <List.Item separated>
                    <div className="row align-items-center">
                        <div className="col-auto">
                            <Avatar>EG</Avatar>
                        </div>
                        <div className="col">
                            Amanda Hunt
                            <small className="d-block item-except text-sm text-muted h-1x">
                                amanda_hunt@example.com
                            </small>
                        </div>
                        <div className="col-auto">
                            <Trash />
                        </div>
                    </div>
                </List.Item>
                <List.Item separated>
                    <div className="row align-items-center">
                        <div className="col-auto">
                            <Avatar>EG</Avatar>
                        </div>
                        <div className="col">
                            Amanda Hunt
                            <small className="d-block item-except text-sm text-muted h-1x">
                                amanda_hunt@example.com
                            </small>
                        </div>
                        <div className="col-auto">
                            <Trash />
                        </div>
                    </div>
                </List.Item>
                <List.Item separated>
                    <div className="row align-items-center">
                        <div className="col-auto">
                            <Avatar>EG</Avatar>
                        </div>
                        <div className="col">
                            Amanda Hunt
                            <small className="d-block item-except text-sm text-muted h-1x">
                                amanda_hunt@example.com
                            </small>
                        </div>
                        <div className="col-auto">
                            <Trash />
                        </div>
                    </div>
                </List.Item>
                <List.Item separated>
                    <div className="row align-items-center">
                        <div className="col-auto">
                            <Avatar>EG</Avatar>
                        </div>
                        <div className="col">
                            Amanda Hunt
                            <small className="d-block item-except text-sm text-muted h-1x">
                                amanda_hunt@example.com
                            </small>
                        </div>
                        <div className="col-auto">
                            <Trash />
                        </div>
                    </div>
                </List.Item>
                <List.Item separated>
                    <div className="row align-items-center">
                        <div className="col-auto">
                            <Avatar>EG</Avatar>
                        </div>
                        <div className="col">
                            Amanda Hunt
                            <small className="d-block item-except text-sm text-muted h-1x">
                                amanda_hunt@example.com
                            </small>
                        </div>
                        <div className="col-auto">
                            <Trash />
                        </div>
                    </div>
                </List.Item>
            </List>
        </Card.Body>
    </Card>
);
