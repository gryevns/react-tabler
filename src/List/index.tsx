import * as React from "react";
import cn from "classnames";

import ListItem from "./ListItem";
import ListGroup from "./ListGroup";
import ListGroupItem from "./ListGroupItem";

export interface ListProps {
    className?: string;
    children: React.ReactNode;
    unstyled?: boolean;
    seperated?: boolean;
    inline?: boolean;
}

function List({ className, children, unstyled, seperated, inline }: ListProps) {
    const classes = cn(
        {
            list: !unstyled,
            "list-unstyled": unstyled,
            "list-seperated": seperated,
            "list-inline": inline,
        },
        className,
    );
    return <ul className={classes}>{children}</ul>;
}

List.Item = ListItem;
List.Group = ListGroup;
List.GroupItem = ListGroupItem;

export default List;
