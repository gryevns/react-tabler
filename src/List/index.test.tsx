import React from "react";
import { shallow } from "enzyme";

import List from "./index";

describe("List", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
            <List>
                <List.Item>An Item</List.Item>
                <List.Item>Another Item</List.Item>
                <List.Item>A third item</List.Item>
            </List>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match snapshot", () => {
        const wrapper = shallow(
            <List unstyled>
                <List.Item>An Item</List.Item>
                <List.Item>Another Item</List.Item>
                <List.Item>A third item</List.Item>
            </List>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
