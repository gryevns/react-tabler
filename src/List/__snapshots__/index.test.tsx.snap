// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<ul
  className="list"
>
  <ListItem>
    An Item
  </ListItem>
  <ListItem>
    Another Item
  </ListItem>
  <ListItem>
    A third item
  </ListItem>
</ul>
`;

exports[`List should match snapshot 2`] = `
<ul
  className="list-unstyled"
>
  <ListItem>
    An Item
  </ListItem>
  <ListItem>
    Another Item
  </ListItem>
  <ListItem>
    A third item
  </ListItem>
</ul>
`;
